units/player.gd

```diff
@@ -9,11 +9,39 @@ const HORIZONTAL_BUFFER = 75
 
 @export var Projectile : PackedScene
 @onready var projectile_spawner = $ProjectileSpawner
+var projectile_effects: Array[ProjectileEffect] = []
+var DoubleLaserEffect : PackedScene = preload("res://projectiles/double_laser.tscn")
+var HugeLaserEffect : PackedScene = preload("res://projectiles/huge_laser.tscn")
+var FastLaserEffect : PackedScene = preload("res://projectiles/fast_laser.tscn")
 
 # Called when the node enters the scene tree for the first time.
 func _ready():
 	Projectile = load("res://units/projectile.tscn")
-
+	_add_test_effects()
+
+func _add_test_effects():
```

Comment:
:wine_glass: :cheese: 

This, an the top-level `projectile_effects` array is not used.

This is for two reasons:

1. The core node type `ProjectileEffect` is not used now, see the comment about using PackedScene instead.
2. Passing _just_ the type `ProjectileEffect` (or a subclass -- HugeLaser, DoubleLaser, etc.) results in an empty array for a reason I truly have no idea about. Otherwise I would keep the array, use a non-instantiated variable, and then use that instead of the `PackedScene`

I think ultimately, a combination of ideas might be the best path forward: A `Globals` script that keeps a list of all our "officially supported" effects as an enum & dictionary of that Enum to a resource path (e.g. `res://projectiles/fast_laser.tscn`), and then maintain the top level list of just the enum, and offload the loading/instansiating to the projectile scene.